lms/djangoapps/instructor_analytics/tests/test_basic.py

```diff
@@ -531,3 +532,54 @@ def test_invalid_module_state(self):
         datarows = list(student_responses(self.course))
         #Invalid module state response will be skipped, so datarows should be empty
         self.assertEqual(len(datarows), 0)
+
+    def test_problem_with_student_answer_and_answers(self):
+        self.course = get_course(CourseKey.from_string('edX/graded/2012_Fall'))
+        self.problem_location = Location('edX', 'graded', '2012_Fall', 'problem', 'H1P2')
+
+        self.create_student()
+        StudentModuleFactory.create(
+            course_id=self.course.id,
+            module_state_key=self.problem_location,
+            student=self.student,
+            grade=0,
+            state=u'{"student_answers":{"problem_id":"student response1"}}'
+        )
+
+        self.submit_and_compare_location = Location('edX', 'graded', '2012_Fall', 'problem', 'H1P3')
+        StudentModuleFactory.create(
+            course_id=self.course.id,
+            module_state_key=self.submit_and_compare_location,
+            student=self.student,
+            grade=0,
+            state=u'{"student_answer": "student response2"}'
+        )
+
+        self.submit_and_compare_location = Location("edX", "graded", "2012_Fall", "problem", 'H1P0')
+        StudentModuleFactory.create(
+            course_id=self.course.id,
+            module_state_key=self.submit_and_compare_location,
+            student=self.student,
+            grade=0,
+            state=u'{"answer": {"problem_id": "123"}}'
+        )
+
+        datarows = list(student_response_rows(self.course))
```

Comment:
Instead of using student_response_rows, we can just use student_responses which my other tests use. Then it becomes
```
datarows = list(student_responses(self.course))
self.assertEqual(datarows[0][-1], u'problem_id=student response1')
self.assertEqual(datarows[1][-1], u'i4x://edX/graded/problem/H1P3=student response2')
```